### ISSUE TITLE
No validation loss is calculated when using test_datasets

### ISSUE THREAD

**Issue by JZacaroli** [axolotl-ai-cloud/axolotl#2781]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Specifying 'test_datasets' in config.yml should calculate and log validation loss on those datasets.

### Current behaviour

No validation loss is computed or logged - only training loss is. It works when you specify `val_set_size`, but not `test_datasets`.

[training_run.txt](https://github.com/user-attachments/files/20687224/training_run.txt)

### Steps to reproduce

With the given config.yml, run `axolotl train config.yml`

### Config yaml

```yaml
base_model: meta-llama/Llama-3.1-8B-Instruct
model_type: LlamaForCausalLM
tokenizer_type: AutoTokenizer

load_in_8bit: false
load_in_4bit: true
strict: false

datasets:
  - path: train.jsonl
    type: alpaca      

dataset_prepared_path:
output_dir: ./outputs/lora-out

# A list of one or more datasets to eval the model with.
# You can use either test_datasets, or val_set_size, but not both.
test_datasets:
  - path: test.jsonl
    ds_type: json
    # You need to specify a split. For "json" datasets the default split is called "train".
    split: train
    type: alpaca
    data_files:
      - test.jsonl
eval_steps: 1

sequence_len: 1000
sample_packing: false
eval_sample_packing: false
pad_to_sequence_len: true

adapter: lora
lora_model_dir:
lora_r: 16
lora_alpha: 16
lora_dropout: 0.05
lora_target_linear: true
lora_fan_in_fan_out:
lora_modules_to_save:
  - embed_tokens
  - lm_head

wandb_project:
wandb_entity:
wandb_watch:
wandb_name:
wandb_log_model:

gradient_accumulation_steps: 1
micro_batch_size: 1
num_epochs: 3
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.0001

train_on_inputs: false
group_by_length: false
bf16: auto
fp16:
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true
s2_attention:

warmup_steps: 0
saves_per_epoch: 1
debug:
deepspeed:
weight_decay: 0.001
fsdp:
fsdp_config:
special_tokens:
   pad_token: <|end_of_text|>
```

### Possible solution

`_configure_save_and_eval_strategy` sets the `eval_strategy` to 'no' if `val_set_size` is 0, which prevents evals from happening.
But `val_set_size` defaults to 0 if it's not specified.

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.11.11

### axolotl branch-commit

main/00cda8c

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by NanoCode012** [comment]
Hey @JZacaroli , see linked PR if that solves it for you

**Comment by JZacaroli** [comment]
Thanks, that works now!

**Comment by NanoCode012** [comment]
> Thanks, that works now!

Thanks for confirming!
